kubectl-cloudflow/verify/blueprint.go

```diff
@@ -0,0 +1,1007 @@
+	imageInStreamletNotInImagesErrors = append(imageInStreamletNotInImagesErrors, checkStreamletImageConsistency(b) ...)
+
+	var streamletNotInImageLabelErrors []BlueprintProblem 
+	streamletNotInImageLabelErrors = append(streamletNotInImageLabelErrors, checkStreamletImageLabelConsistency(b) ...)
+
+	// get all streamlet descriptors for all images
+	var streamletDescriptors []StreamletDescriptor
+	for _, desc := range b.streamletDescriptorsPerImage {
+		streamletDescriptors = append(streamletDescriptors, desc ...)
+	}
+
+	var emptyStreamletDescriptorsProblem *EmptyStreamletDescriptors
+	if len(streamletDescriptors) == 0 {
+		emptyStreamletDescriptorsProblem = &EmptyStreamletDescriptors{}
+	}
+
+	var newStreamlets []StreamletRef
+	var verifiedStreamlets []VerifiedStreamlet
+
+	for _, ref := range b.streamlets {
+		newStreamlets = append(newStreamlets, ref.verify(streamletDescriptors))
+	}
+
+	for _, streamlet := range newStreamlets {
+		if streamlet.verified != nil {
+			verifiedStreamlets = append(verifiedStreamlets, *streamlet.verified)
+		}
+	}
+
+	var newConnections []StreamletConnection
+	var verifiedConnections []VerifiedStreamletConnection
+	var connectionVerifyProblems []BlueprintProblem
+
+	for _, con := range b.connections {
+		newConnection := con.verify(verifiedStreamlets)
+		connectionVerifyProblems = append(connectionVerifyProblems, newConnection.problems...)
+		newConnections = append(newConnections, newConnection)
+	}
+
+	for _, verCon := range newConnections {
+		if verCon.verified != nil {
+			verifiedConnections = append(verifiedConnections, *verCon.verified)
+		}
+	}
+
+	_, duplicatesProblem := b.verifyNoDuplicateStreamletNames(newStreamlets)
+
+	portNameProblems = b.verifyPortNames(streamletDescriptors)
+	configParameterProblems = b.verifyConfigParameters(streamletDescriptors)
+	volumeMountProblems = b.verifyVolumeMounts(streamletDescriptors)
+
+	verifiedConnections, conProblems := b.verifyUniqueInletConnections(verifiedConnections)
+
+	for _, conProblem := range conProblems {
+		illegalConnectionProblems = append(illegalConnectionProblems, conProblem)
+	}
+
+	var inletProblems []BlueprintProblem
+
+	inletProblems = append(inletProblems, illegalConnectionProblems...)
+
+	for _, newCon := range newConnections {
+		for _, p := range newCon.problems {
+			_, ok := p.(InletProblem)
+
+			if ok {
+				inletProblems = append(inletProblems, p)
+			}
+		}
+	}
+	var globalProblems []BlueprintProblem
+
+	_, inletConProblems := b.verifyInletsConnected(verifiedStreamlets, verifiedConnections)
+
+	for _, inletConProblem := range inletConProblems {
+		var filteredUnconnectedInlets []UnconnectedInlet
+		filteredUnconnectedInlets = filterUnconnectedInlets(inletProblems, inletConProblem.unconnectedInlets)
+
+		if len(filteredUnconnectedInlets) > 0 {
+			unconnectedInletProblems = append(unconnectedInletProblems, UnconnectedInlets{unconnectedInlets: filteredUnconnectedInlets})
+		}
+	}
+
+	if emptyStreamletsProblem != nil {
```

Comment:
I'm reading this function, and this is just me thinking out loud, but I cannot make up my mind if the function is too long and each problem type should be added at the point they are found, or keep it like this for readability.

Reply:
This happens when you translate compact Scala code to code. It is verbose I agree but this is a first attempt more focused on getting things working at this point of time and making things readable as you mentioned to make sure that porting the code was successful.

Reply:
@skonto Yea, I totally understand that. On a larger point, I think it would be useful to leave things like this as-is, and maybe comment something like "TODO: REFACTOR", to move ahead. Anyone could pick that up some slow day or a day you just want to do some lighter work.